src/containers/RedashContainer/index.jsx

```diff
@@ -1,50 +1,24 @@
+/* eslint-disable camelcase */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { withStyles } from '@material-ui/core/styles';
 import ChartJsWrapper from '../../vendor/chartJs/ChartJsWrapper';
-import fetchJson from '../../utils/fetchJson';
+import { fetchJson } from '../../vendor/requests';
 import { withErrorBoundary } from '../../vendor/errors';
-import { Log } from '../../vendor/logs';
-
-/* eslint-disable camelcase */
+import { selectFrom } from '../../vendor/vectors';
 
-const sortByDate = (a, b) =>
-  new Date(b.submission_date) - new Date(a.submission_date);
-const dataToChartJSformat = data =>
-  data.map(({ submission_date, value }) => ({
-    x: submission_date,
-    y: value,
-  }));
 const telemetryDataToDatasets = (data, dataKeyIdentifier) => {
-  const queryResulset = data.query_result;
-  // Separate data points into buckets
-  const buckets = queryResulset.data.rows.reduce((result, datum) => {
-    const key = datum[dataKeyIdentifier];
-
-    if (!key) {
-      // XXX: Make it a custom error, catch it and display a UI message
-      Log.error(
-        'Check the Redash data and determine what is the key used to categorize the data.'
-      );
-    }
-
-    if (!result[key]) {
-      // eslint-disable-next-line no-param-reassign
-      result[key] = [];
-    }
-
-    result[key].push(datum);
-
-    return result;
-  }, {});
-  const datasets = Object.keys(buckets).map(key => {
-    const datum = buckets[key];
```

Comment:
`buckets` was built from an object (`{}`) which is subsequently converted to a sequence {key, datum} pairs above; the intermediate `buckets` is not needed. `.groupBy()` provides this sequence directly.